session/autocharthandler.go

```diff
@@ -23,9 +24,33 @@ func (handler *AutoChartHandler) Instance(id string) ObjectHandlerInstance {
 
 // GetObject implement ObjectHandler interface
 func (instance *AutoChartInstance) SetObjectAndEvents(sessionState *State, actionState *action.State, obj *enigmahandlers.Object, genObj *enigma.GenericObject) {
+	instance.GetObjectLayout(sessionState, actionState, obj)
+	handleAutoChart(sessionState, actionState, genObj, obj)
+}
+
+// GetObjectLayout for auto-chart main object
+func (instance *AutoChartInstance) GetObjectLayout(sessionState *State, actionState *action.State, obj *enigmahandlers.Object) {
 	sessionState.QueueRequest(func(ctx context.Context) error {
-		return GetObjectLayout(sessionState, actionState, obj)
-	}, actionState, true, "")
+		enigmaObject, ok := obj.EnigmaObject.(*enigma.GenericObject)
```

Comment:
Assume this is a different object than `genObj *enigma.GenericObject` in the call to `SetObjectAndEvents`?

Reply:
auto-chart main object will always be *enigma.GenericObject

Reply:
ah, understand the question now, no should be the same, logic was copied from the generic version, I could probably send in genObj instead, yes

Reply:
Doesn't really matter. But that means this code and the error check are not strictly needed.

Reply:
you're right, didn't check it through for redundant things.